kie-maven-plugin/src/main/java/org/kie/maven/plugin/BuildMojo.java

```diff
@@ -247,72 +251,73 @@ private void saveFile(MemoryFileSystem mfs, String fileName) throws MojoFailureE
         }
     }
 
-    private void shareKieObjectsWithMap(InternalKieModule kModule) {
-        Optional<Map<String, Object>> optionalKieMap = getKieMap();
-        if (optionalKieMap.isPresent()) {
-            KieMetaInfoBuilder builder = new KieMetaInfoBuilder(kModule);
-            KieModuleMetaInfo modelMetaInfo = builder.getKieModuleMetaInfo();
+    private String getCompilationID(Map<String, Object> optionalKieMap) {
```

Comment:
@jhrcek we can't test the kie-map here (in this prj) because is always null when the kie-maven-plugin is invoked without the compiler in the kie-wb-common

Reply:
@desmax74 The reason why it can't be tested here is because it's part of one big monolithic function.
You can extract the logic that modifies the input kieMap into separate package private method and test that method separately. Something like
```
               if (optionalKieMap.isPresent()) {
                    Map<String, Object> kieMap = optionalKieMap.get();
                    addStuffToKieMap(kieMap, kModule);
                } else {
                    getLog().info("Kie Map not present");
                }
...

   // create new package private so it can be tested separately
    void addStuffToKieMap(...) {
                    String compilationID = getCompilationID(kieMap);
                    shareKieObjectsWithMap(kModule, compilationID, kieMap);
                    shareStoreWithMap(kModule.getModuleClassLoader(), compilationID, kieMap);
                    shareTypesMetaInfoWithMap(kModule, compilationID, kieMap);
     }
```
There is a lot of logic and branching in this parameter-processing code and it should be unit tested. This way we'll cover all possible cases easily in fast-running tests, which you be much harder to achieve in integration tests.

Reply:
@jhrcek Moved the logic used by the compiler into CompilerHelper class